libsql/internal/http/driver.go

```diff
@@ -67,40 +99,131 @@ func (c *conn) Begin() (driver.Tx, error) {
 	return nil, fmt.Errorf("begin method not implemented")
 }
 
-func convertArgs(args []driver.NamedValue) params {
+func (c *conn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
+	rs, err := callSqld(ctx, c.url, c.jwt, query, args)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := assertNoResultWithError(rs, query); err != nil {
+		return nil, err
+	}
+
+	return &result{0, 0}, nil
+}
+
+func (c *conn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	rs, err := callSqld(ctx, c.url, c.jwt, query, args)
+	if err != nil {
+		return nil, err
+	}
+	return &rows{rs, 0, 0}, nil
+}
+
+func assertNoResultWithError(resultSets []httpResults, query string) error {
+	for _, result := range resultSets {
+		if result.Error != nil {
+			return fmt.Errorf("failed to execute SQL: %s\n%s", query, result.Error.Message)
+		}
+		if result.Results == nil {
+			return fmt.Errorf("no results for statement")
+		}
+	}
+	return nil
+}
+
+func callSqld(ctx context.Context, url string, jwt string, sql string, sqlParams []driver.NamedValue) ([]httpResults, error) {
+	parameters, err := convertArgs(sqlParams)
+	if err != nil {
+		return nil, err
+	}
+
+	rawReq, err := generatePostBody(sql, parameters)
+	if err != nil {
+		return nil, err
+	}
+
+	reqBody, err := json.Marshal(rawReq)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(reqBody))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+  if len(jwt) > 0 {
+    req.Header.Set("Authorization", "Bearer " + jwt)
+  }
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		var err_response struct {
+			Message string `json:"error"`
+		}
+		if err := json.Unmarshal(body, &err_response); err != nil {
+			return nil, fmt.Errorf("failed to execute SQL: %s", sql)
+		}
+		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, err_response.Message)
+	}
+
+	var results []httpResults
+
+	if err := json.Unmarshal(body, &results); err != nil {
+		return nil, err
+	}
+
+	if results[0].Error != nil {
+		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, results[0].Error.Message)
+	}
+	if results[0].Results == nil {
+		return nil, fmt.Errorf("no results for SQL: %s", sql)
+	}
+	return results, nil
+}
+
+func convertArgs(args []driver.NamedValue) (params, error) {
```

Comment:
Just by moving this function below ExecContext and QueryContext you screwed up the whole diff. Please don't do this in the future. If you really can't restrain yourself from moving things around then please do it in a separate commit. Otherwise the whole diff is a complete mess that's very unpleasant to review.

Reply:
My bad